orthopairs/src/main/java/org/reactome/release/orthopairs/OrthopairFileGenerator.java

```diff
@@ -0,0 +1,93 @@
+        }
+    }
+
+    // Create target species gene-protein mapping file
+    public static void createSpeciesGeneProteinFile(String speciesKey, String targetGeneProteinMappingFilename, JSONObject speciesJSON, Map<String,Set<String>> speciesGeneProteinMap) throws IOException {
+
+        logger.info("\tGenerating " + targetGeneProteinMappingFilename);
+        File targetGeneProteinMappingFile = new File (targetGeneProteinMappingFilename);
+        if (targetGeneProteinMappingFile.exists()) {
+            targetGeneProteinMappingFile.delete();
+        }
+
+        boolean altIdMappingExists = false;
+        Map<String, Set<String>> altIdToEnsemblMap = new HashMap<>();
+        if (speciesJSON.get("alt_id_file") != null) {
+            logger.info("\tAlternate ID-Ensembl ID mapping required");
+            altIdToEnsemblMap = AlternateIdMapper.getAltIdMappingFile(speciesKey, speciesJSON.get("alt_id_file").toString());
+            altIdMappingExists = true;
+        }
+        targetGeneProteinMappingFile.createNewFile();
+
+        List<String> targetGeneProteinLines = new ArrayList<>();
+        for (String targetGeneId : speciesGeneProteinMap.keySet()) {
+            String[] geneSplit = targetGeneId.split("=");
+            String geneSource = geneSplit[0];
+            String geneId = geneSplit[geneSplit.length - 1];
+            Set<String> targetProteinIds = speciesGeneProteinMap.get(targetGeneId);
+            targetProteinIds.remove("LDO");
+            List<String> cleanTargetProteinIds = new ArrayList<>();
+            for (String targetProteinId : speciesGeneProteinMap.get(targetGeneId)) {
+                cleanTargetProteinIds.add(targetProteinId.split("=")[1]);
+            }
+            if (cleanTargetProteinIds.size() > 0) {
+                Collections.sort(cleanTargetProteinIds);
+                if (!geneSource.startsWith("Ensembl") && altIdMappingExists) {
+                    Set<String> ensemblGeneIds = altIdToEnsemblMap.get(geneId);
+                    if (ensemblGeneIds != null) {
+                        for (String ensemblGeneId : altIdToEnsemblMap.get(geneId)) {
+                            String geneProteinLine = ensemblGeneId + "\t" + String.join(" ", cleanTargetProteinIds) + "\n";
+                            targetGeneProteinLines.add(geneProteinLine);
+                        }
+                    }
+                } else {
+                    String geneProteinLine = geneId + "\t" + String.join(" ", cleanTargetProteinIds) + "\n";
+                    targetGeneProteinLines.add(geneProteinLine);
+                }
+            }
+        }
+        Collections.sort(targetGeneProteinLines);
+        for (String targetGeneProteinLine : targetGeneProteinLines) {
+            Files.write(Paths.get(targetGeneProteinMappingFile.getPath()), targetGeneProteinLine.getBytes(), StandardOpenOption.APPEND);
```

Comment:
Like above, the `targetGeneProteinMappingFile` String can be used for the argument to Paths.get()